extension.js

```diff
@@ -330,10 +405,23 @@ class Extension {
         this.overview_shown = Main.overview.connect('shown', this._on_overview_shown.bind(this));
         this.overview_hiding = Main.overview.connect('hiding', this._on_overview_hiding.bind(this));
 
-        this.workareas_changed = global.display.connect_after('workareas-changed', this._dock_refresh.bind(this));
-        this.window_created = global.display.connect_after('window-created', this._dock_refresh.bind(this));
-        this.restacked = global.display.connect_after('restacked', this._dock_refresh.bind(this));
-        this.size_changed = global.window_manager.connect_after('size-changed', this._dock_refresh.bind(this));
+        // refresh again after overview is hidden or else the dock size is messed up
+        // this is only noticeable with auto hide enabled
+        // todo: find a better way to fix this issue
+        this.overview_hidden = Main.overview.connect('hidden', this._dock_refresh.bind(this));
```

Comment:
Having the dock refresh twice in a row seems unnecessary, and honestly I think the dock might be refreshing too often. This is the only way I could figure out how to fix the issue of the dock being the wrong size after exiting the overview though. If someone wants to replicate the issue, just open and close the overview with autohide enabled and you will notice that the window overlap causes the dock to hide when it is about 12 pixels above the dock, instead of when it actually touches the dock. This is because the dock gets set to the wrong size during the refresh that happens when exiting the overview.